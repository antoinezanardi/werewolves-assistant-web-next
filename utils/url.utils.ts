function removeTrailingSlashes(url: string): string {
  return url.replace(/\/+$/u, "");
}

export { removeTrailingSlashes };