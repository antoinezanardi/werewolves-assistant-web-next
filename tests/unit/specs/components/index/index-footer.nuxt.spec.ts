import type { mount } from "@vue/test-utils";

import { IndexFooter } from "#components";
import { mountSuspendedComponent } from "~/tests/unit/utils/mount.utils";

describe("Index Footer Component", () => {
  let wrapper: ReturnType<typeof mount<typeof IndexFooter>>;

  beforeEach(async() => {
    wrapper = await mountSuspendedComponent(IndexFooter);
  });

  it("should render component and match snapshot when mounted.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Contact Me Button", () => {
    it("should render contact me with translated text button when rendered.", () => {
      const contactMeButton = wrapper.find("[aria-label='Contact me']");

      expect(contactMeButton.text()).toBe("Contact me");
    });
  });
});