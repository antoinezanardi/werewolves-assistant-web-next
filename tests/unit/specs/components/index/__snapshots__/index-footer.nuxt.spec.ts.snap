// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index Footer Component > should render component and match snapshot when mounted. 1`] = `
"<div class="row">
  <div class="col-6"><a aria-label="Contact me" class="p-button p-button-secondary" href="mailto: antoine.zanardi@epitech.eu"><span class="fa fa-envelope me-2"></span> Contact me</a></div>
  <div class="col-6 d-flex justify-content-end text-right"><a aria-label="This project is open source" class="p-button p-button-help" href="https://www.github.com/antoinezanardi/werewolves-assistant-web-next" rel="noopener noreferrer" target="_blank"><span class="fa-github fab me-2"></span> This project is open source</a></div>
</div>"
`;
